src/error-table.rkt

```diff
@@ -0,0 +1,49 @@
+#lang racket
+
+(require "points.rkt" "syntax/types.rkt" "core/localize.rkt" "timeline.rkt")
+
+(provide group-errors)
+
+(define (zip l1 l2) (map list l1 l2))
+
+(define (group-errors expr pcontext)
+  (define points (for/list ([(pt _) (in-pcontext pcontext)]) pt))
+  (define local-errors (car (compute-local-errors (list expr) (*context*))))
+
+  (define max-per-point (make-hash))
+  (define errpts-per-subexpr (make-hash))
+
+  (for ([pt points])
```

Comment:
I think this code is way overly verbose for what it's trying to do, and that's because you're stuck going per-subexpr whereas what you want to do is per-expr. Let's do something like this...

First, get the list of subexpressions and, for every point, the local error of each subexpression. Note the use of `flip-lists`, this transposes a matrix, defined in `common.rkt`:

```
(match-define (cons subexprs pt-errorss)
  (flip-lists (hash->list (car (compute-local-errors (list expr) (*context*))))))
```

Now for every point we need which subexpression has the worst error (if bigger than 2). To do that we pair up subexpression to error on that point, and pick out the pair with the worst error using `argmax`:

```
(define pt-worst-subexprs
  (for/list ([pt-errors (in-list pt-errorss)])
    (match-define (cons worst-subexpr pt-max-error)
      (argmax cdr (map cons subexprs pt-errors)))
    (if (> pt-max-error 2) worst-subexpr #f)))
```

Now we need to know how many times each subexpression appears in the list. One way to do it: group by identity, count size of each group:

```
(for/hash ([group (in-list (group-by identity pt-worst-subexprs))])
  (values (first group) (length group)))
```